src/Data/Permissions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace App\Data;
+
+use App\Data\Entity\User;
+
+class Permissions {
+
+    /**
+     * @var Authenticator
+     */
+    protected $auth;
+
+    /**
+     * @var User
+     */
+    protected $user;
+
+    public function __construct(Authenticator $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    public function setUser(User $user)
```

Comment:
Pretty sure that the constructor should take the user, not be part of a separate method. If the permissions accidentally gets shared, the user context could switch unexpectedly as a side effect.